datarecording/execlog.go

```diff
@@ -0,0 +1,78 @@
+package datarecording
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+// Struct ExecInfo is feed to DataRecorder
+type ExecInfo struct {
+	Property string
+	Value    string
+}
+
+// Records program execution
+type execRecorder struct {
+	// db        *sql.DB
+	tablename string
+	recorder  DataRecorder
+	entries   []ExecInfo
+}
+
+// Write log current execution
+func (e *execRecorder) Write() {
+	currentTime := time.Now()
+	startTime := currentTime.Format("2006-01-02 15:04:05")
+	timeEntry := ExecInfo{"Start Time", startTime}
```

Comment:
This is a BIG problem here. The start time is recorded in the Write function and the write function is part of the atexit function call, which means the start time is only recorded at the end of the simulation.

Reply:
Start time should be recorded in the new function.